pint_server/app.py

```diff
@@ -126,6 +187,12 @@ def json_to_xml(json_obj, collection_name, element_name):
 
 
 def get_formatted_dict(obj, extra_attrs=None, exclude_attrs=None):
+    # make exclude attrs an empty list if not specified so that we
+    # don't have to check for it being non-None on every iteration
+    # below
+    if exclude_attrs is None:
+        exclude_attrs = []
```

Comment:
Nice!

Reply:
Modern Python byte code optimisation probably eliminates any performance penalty by detecting the loop invariant and moving resolution of it to before entering the loop code, but changing the exclude_attrs to a list here removes the need to include the test in the written code, which makes it a little easier to read. ;-)